packages/kafka-iam-auth/src/index.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable functional/no-let */
+/* eslint-disable no-constant-condition */
+export * from "./constants.js";
+export * from "./create-authenticator.js";
+export * from "./create-mechanism.js";
+export * from "./create-payload.js";
+export * from "./create-sasl-authentication-request.js";
+export * from "./create-sasl-authentication-response.js";
+import { Consumer, Kafka, KafkaMessage } from "kafkajs";
+import { ConsumerConfig, logger } from "pagopa-interop-commons";
+import { createMechanism } from "./create-mechanism.js";
+
+export const DEFAULT_AUTHENTICATION_TIMEOUT = 60 * 60 * 1000;
+export const REAUTHENTICATION_THRESHOLD = 20 * 1000;
+
+const errorTypes = ["unhandledRejection", "uncaughtException"];
+const signalTraps = ["SIGTERM", "SIGINT", "SIGUSR2"];
+
+const handleExit = (consumer: Consumer): void => {
+  errorTypes.forEach((type) => {
+    process.on(type, async (e) => {
+      try {
+        logger.info(`process.on ${type}`);
+        logger.error(e);
+        await consumer.disconnect().finally(() => {
+          logger.info("Consumer disconnected");
+          process.exit(0);
+        });
+        process.exit(0);
+      } catch (_) {
+        process.exit(1);
+      }
+    });
+  });
+
+  signalTraps.forEach((type) => {
+    process.once(type, async () => {
+      try {
+        await consumer.disconnect().finally(() => {
+          logger.info("Consumer disconnected");
+          process.exit(0);
+        });
+      } finally {
+        process.kill(process.pid, type);
+      }
+    });
+  });
+};
+
+const initConsumer = async (
+  config: ConsumerConfig,
+  topic: string,
```

Comment:
Maybe we could use an array of topics to avoid the creation of an array during the `consumer.subscribe`. In the future, we may have more than a single-topic subscription (not sure about this requirement).